motor/discovery/gcp/mql_assets.go

```diff
@@ -0,0 +1,174 @@
+package gcp
+
+import (
+	"errors"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/rs/zerolog/log"
+	"go.mondoo.com/cnquery"
+	"go.mondoo.com/cnquery/llx"
+	"go.mondoo.com/cnquery/motor"
+	"go.mondoo.com/cnquery/motor/asset"
+	"go.mondoo.com/cnquery/motor/discovery/common"
+	"go.mondoo.com/cnquery/motor/platform"
+	"go.mondoo.com/cnquery/motor/providers"
+	gcpprovider "go.mondoo.com/cnquery/motor/providers/google"
+	"go.mondoo.com/cnquery/mql"
+	"go.mondoo.com/cnquery/resources"
+	resource_pack "go.mondoo.com/cnquery/resources/packs/gcp"
+)
+
+const RegionLabel string = "mondoo.com/region"
+
+type MqlDiscovery struct {
+	rt *resources.Runtime
+}
+
+func NewMQLAssetsDiscovery(provider *gcpprovider.Provider) (*MqlDiscovery, error) {
+	m, err := motor.New(provider)
```

Comment:
I am not yet sure if this will work if you pass credentials to authenticate, I had the same trouble with Azure.

Basically, the secret in the provider is unresolved and you need to resolve this by using the credentials query that the resolver gets passed.

This is the same reason why we use `motor.NewMotorConnection` in the resolver directly instead of `motor.New`.

Tldr: Should probably be using `motor.NewMotorConnection` here

Reply:
i'm not sure i understand -- can you give an example of what you mean when you say "pass credentials to authenticate", like the command you're using/thinking of?

Reply:
I can describe that. I have this inventory file for GCP:
```yaml
apiVersion: v1
kind: Inventory
metadata:
  name: inventory
spec:
  assets:
    - name: cool-stuff
      connections:
        - backend: 13
          credentials:
            - secret_id: storage/random-bucket2/foo
              type: 6
              secret_encoding: 3
          options:
          discover:
            targets:
              - compute-images
              - projects
  vault:
    name: gcp-berglas
    type: gcp-berglas
    options:
      project_id: mondoo-dev-262313
```

That breaks with discovering assets because the secret isn't resolved by default

Reply:
👍  thanks for the example!